djstripe/models/base.py

```diff
@@ -349,6 +349,13 @@ def _stripe_object_to_record(
                     #  sub-classes CharField, is that intentional?
                     field_data = ""
 
+                # so that DecimalField max_digits and decimal_places are
+                # respected in the returned instance
+                if isinstance(field, models.DecimalField) and field_data:
+                    field_data = field.to_python(
+                        f"{field_data:0.{field.decimal_places}f}"
```

Comment:
This doesn’t seem right. Is the data float data at that point? What type is the original field_data being referenced here?

Reply:
Original data, `field_data`,  comes from a previous block and it is `float` and not `Decimal` at this point so converting it to `Decimal`. The other option would be to call `instance.refresh_from_db()` in `sync_from_stripe_data` but that would update **all** fields which would be unnecessary.